gensim/test/test_ensemblelda.py

```diff
@@ -0,0 +1,419 @@
+            passes=passes, num_models=num_models, random_state=random_state,
+            memory_friendly_ttda=False, topic_model_class=LdaModel,
+        )
+
+    def check_ttda(self, ensemble):
+        """tests the integrity of the ttda of any ensemble"""
+        self.assertGreater(len(ensemble.ttda), 0)
+        a = ensemble.ttda.sum(axis=1)
+        b = np.ones(len(ensemble.ttda)).astype(np.float32)
+        np.testing.assert_allclose(a, b, rtol=1e-04)
+
+    def test_eLDA(self):
+        # given that the random_state doesn't change, it should
+        # always be 2 detected topics in this setup.
+        self.assertEqual(self.eLDA.stable_topics.shape[1], len(common_dictionary))
+        self.assertEqual(len(self.eLDA.ttda), num_models * num_topics)
+        self.check_ttda(self.eLDA)
+
+        # reclustering shouldn't change anything without
+        # added models or different parameters
+        self.eLDA.recluster()
+
+        self.assertEqual(self.eLDA.stable_topics.shape[1], len(common_dictionary))
+        self.assertEqual(len(self.eLDA.ttda), num_models * num_topics)
+        self.check_ttda(self.eLDA)
+
+        # compare with a pre-trained reference model
+        reference = EnsembleLda.load(datapath('ensemblelda'))
+        np.testing.assert_allclose(self.eLDA.ttda, reference.ttda, rtol=rtol)
+        # small values in the distance matrix tend to vary quite a bit around 2%,
+        # so use some absolute tolerance measurement to check if the matrix is at least
+        # close to the target.
+        atol = reference.asymmetric_distance_matrix.max() * 1e-05
+        np.testing.assert_allclose(
+            self.eLDA.asymmetric_distance_matrix,
+            reference.asymmetric_distance_matrix, atol=atol,
+        )
+
+    def test_clustering(self):
+        # the following test is quite specific to the current implementation and not part of any api,
+        # but it makes improving those sections of the code easier as long as sorted_clusters and the
+        # cluster_model results are supposed to stay the same. Potentially this test will deprecate.
+
+        reference = EnsembleLda.load(datapath('ensemblelda'))
+        cluster_model_results = deepcopy(reference.cluster_model.results)
+        sorted_clusters = deepcopy(reference.sorted_clusters)
+        stable_topics = deepcopy(reference.get_topics())
+
+        # continue training with the distance matrix of the pretrained reference and see if
+        # the generated clusters match.
+        reference.asymmetric_distance_matrix_outdated = True
+        reference.recluster()
+
+        self.assert_cluster_results_equal(reference.cluster_model.results, cluster_model_results)
+        self.assertEqual(reference.sorted_clusters, sorted_clusters)
+        np.testing.assert_allclose(reference.get_topics(), stable_topics, rtol=rtol)
+
+    def test_not_trained(self):
+        # should not throw errors and no training should happen
+
+        # 0 passes
```

Comment:
Please make each of these a separate test case. Instead of the one-line comment, name each test case, e.g.

- test_not_trained_0_passes
- test_not_trained_no_corpus

Reply:
I tried to clean up some other test case names too, and there are loads of improved test variable names to improve test readability